os/arch/arm/src/amebasmart/amebasmart_idle.c

```diff
@@ -129,76 +121,67 @@ But in case that happens, we have to introduce some strategy to handle that
-							if (up_get_secondary_cpu_state(1) == CPU_WAKE_FROM_SLEEP) {
-								pmdbg("Secondary core just woke from PG sleep!\n");
-								goto RESLEEP_2;
-							}
-							/* CG flow */
-						} else {
-							if (!check_wfi_state(1)) {
-								goto EXIT;
-							}
-						}
-#endif
-						/* Interrupt source will wake cpu up, just leave expected idle time as 0
-						Enter sleep mode for AP */
-						configPRE_SLEEP_PROCESSING(xModifiableIdleTime);
-						/* When wake from pg, arm timer has been reset, so a new compare value is necessary to
-						trigger an timer interrupt */
-						if (pmu_get_sleep_type() == SLEEP_PG) {
+					/*PG flow */
+					if (pmu_get_sleep_type() == SLEEP_PG) {
+						/* CPU1 is in task schedular, tell CPU1 to enter hotplug */
+						if (up_get_secondary_cpu_state(1) == CPU_RUNNING) {
+							/* CPU1 may in WFI idle state. Wake it up to enter hotplug itself */
+							up_irq_enable();
+							arm_cpu_sgi(GIC_IRQ_SGI4, (1 << 1));
+							arm_arch_timer_int_mask(0);
 							up_timer_enable();
-							arm_arch_timer_set_compare(arm_arch_timer_count() + 50000);
+							DelayUs(100);
+							goto RESLEEP_1;
+						}
+						/* CPU1 just come back from pg, so can't sleep here */
+						if (up_get_secondary_cpu_state(1) == CPU_WAKE_FROM_SLEEP) {
+							pmdbg("Secondary core just woke from PG sleep!\n");
+							goto RESLEEP_2;
+						}
+						/* CG flow */
+					} else {
+						if (!check_wfi_state(1)) {
+							goto EXIT;
 						}
-						arm_arch_timer_int_mask(0);
-						configPOST_SLEEP_PROCESSING(xModifiableIdleTime);
 					}
-					else {
-						/* power saving when idle*/
-						arm_arch_timer_int_mask(0);
-						__asm(" DSB");
-						__asm(" WFI");
-						__asm(" ISB");
+#endif
+					/* Interrupt source will wake cpu up, just leave expected idle time as 0
+					Enter sleep mode for AP */
+					configPRE_SLEEP_PROCESSING(xModifiableIdleTime);
+					/* When wake from pg, arm timer has been reset, so a new compare value is necessary to
+					trigger an timer interrupt */
+					if (pmu_get_sleep_type() == SLEEP_PG) {
+						up_timer_enable();
+						arm_arch_timer_set_compare(arm_arch_timer_count() + 50000);
 					}
+					arm_arch_timer_int_mask(0);
+					configPOST_SLEEP_PROCESSING(xModifiableIdleTime);
+				}
+				else {
+					/* power saving when idle*/
+					arm_arch_timer_int_mask(0);
+					__asm(" DSB");
+					__asm(" WFI");
+					__asm(" ISB");
+				}
 #ifdef CONFIG_SMP
 EXIT:
 #endif				
-					/* Re-enable interrupts and sys tick*/
-					up_irq_enable();
-				}
-				/* Note: Wakeup from sleep, change the state back to PM_NORMAL 
-				   At this point, we do not need to do anything, as the
-				   wakeup callback handler will invoke pm_activity()
-				*/
+				/* Re-enable interrupts and sys tick*/
+				up_irq_enable();
+			}
+			/* Note: Wakeup from sleep, change the state back to PM_NORMAL 
+				At this point, we do not need to do anything, as the
+				wakeup callback handler will invoke pm_activity()
```

Comment:
Since we have changed the logic flow for state transition after wakeup, could you please help to revise the comment here abit?
ie. perhaps like "invoke pm_activity" -> "handle the state transition"

Reply:
Modified the comment based on new PM logic